agent/consul/replication.go

```diff
@@ -0,0 +1,131 @@
+package consul
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	"golang.org/x/time/rate"
+)
+
+const (
+	// replicationMaxRetryBackoff is the maximum number of seconds to wait between
+	// failed blocking queries when backing off.
+	replicationMaxRetryBackoff = 256
+)
+
+type ReplicatorConfig struct {
+	Name            string
+	ReplicateFn     ReplicatorFunc
+	Rate            int
+	Burst           int
+	MaxRetryBackoff int
```

Comment:
nit: should this be a `time.Duration`? That's more consistent with other internal APIs IMO. If not for some reason maybe we should at least name/comment that it's an integer number of seconds?

Reply:
This is just how we do it for other replication routines. I am not against changing that though.